### PR TITLE
 docs: update route.md to use openapi-v3 

### PR DESCRIPTION
**Pull request by shimks**
Please review the 2nd commit :)

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: docs/site/Routes.md [hunk 1]
```diff
@@ -10,83 +10,120 @@ summary:
 
 ## Overview
 
-A `Route` is the mapping between your API specification and an Operation (JavaScript implementation). It tells LoopBack which function to `invoke()` given an HTTP request.
+A `Route` is the mapping between your API specification and an Operation. It
```
Comment (raymondfeng) [inline docs/site/Routes.md]: Maybe it's better to replace `an Operation` with `a handler function in JavaScript, such as a controller method`

Comment (shimks) [reply docs/site/Routes.md]: I've expanded upon the definition under `Operations` header with your example

File: docs/site/Routes.md [hunk 2]
```diff
@@ -10,83 +10,120 @@ summary:
...
+* With an OpenAPI specification object
+* Using partial OpenAPI spec fragments with the `Route` constructor
+* Using route decorators on controller methods
 
 ### Declaring REST Routes with API specifications
 
-Below is an example [Open API Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object) that defines the same operation as the example above. This a declarative approach to defining operations. The `x-operation` field in the example below references the handler JavaScript function for the API operation, and should not be confused with `x-operation-name`, which is a string for the Controller method name.
+Below is an example
+[Open API Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#oasObject)
+that defines the same operation as the example above. This a declarative
```
Comment (dhmlau) [inline docs/site/Routes.md]: `This a declarative` -> `This is a declarative`?

File: docs/site/Routes.md [hunk 3]
```diff
@@ -10,83 +10,120 @@ summary:
...
+approach to defining operations. The `x-operation` field in the example below
```
Comment (dhmlau) [inline docs/site/Routes.md]: `to defining operations` -> `to define operations`?

Comment (shimks) [reply docs/site/Routes.md]: I might change it to `This is the declarative approach to defining operations`, but what's up there (aside from `is` missing) looks good to me unless I'm horribly off.

File: docs/site/Routes.md [hunk 4]
```diff
@@ -10,83 +10,120 @@ summary:
...
-- Using route decorators on controller methods
+
+* With an OpenAPI specification object
+* Using partial OpenAPI spec fragments with the `Route` constructor
+* Using route decorators on controller methods
 
 ### Declaring REST Routes with API specifications
 
-Below is an example [Open API Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object) that defines the same operation as the example above. This a declarative approach to defining operations. The `x-operation` field in the example below references the handler JavaScript function for the API operation, and should not be confused with `x-operation-name`, which is a string for the Controller method name.
+Below is an example
```
Comment (virkt25) [inline docs/site/Routes.md]: an example **of an**

File: docs/site/Routes.md [hunk 5]
```diff
@@ -143,7 +185,8 @@ app.start();
 
 ## Invoking operations using Routes
 
-This example breaks down how `Sequences` determine and call the matching operation for any given request.
+This example breaks down how `Sequences` determine and call the matching
```
Comment (virkt25) [inline docs/site/Routes.md]: `This example breaks down how a [Sequence](LINK TO SEQUENCE DOC) determines and calls the matching operation for any request.`

File: docs/site/Routes.md [hunk 6]
```diff
@@ -10,83 +10,120 @@ summary:
...
+plain JavaScript/TypeScript functions (like http handler functions) or as
+methods in [Controllers](Controllers.md).
 
-```js
+```ts
 // greet is a basic operation
-function greet(name) {
-  return `hello ${name}`;
+function greet(name: string) {
+  return `hello ${name || user}`;
```
Comment (b-admike) [inline docs/site/Routes.md]: where is `user` coming from?

Comment (shimks) [reply docs/site/Routes.md]: remnant of experimental code, fixed

**Comment by shimks** [comment]
@slnode test please

**Pull request merged**